bundles/framework/layerlist/view/LayerViewTabs/LayerList/LayerCollapse/LayerCollapsePanel.jsx

```diff
@@ -30,31 +50,132 @@ const StyledEditGroup = styled.span`
     return (
-        <StyledListItem>
-            <Layer key={model.getId()} {...itemProps} />
-        </StyledListItem>
+            <StyledListItem>
+                <Layer key={model.getId()}  {...itemProps} />
+            </StyledListItem>
     );
 };
+
 renderLayer.propTypes = {
     model: PropTypes.any,
     even: PropTypes.any,
     selected: PropTypes.any,
     controller: PropTypes.any
 };
 
-const onToolClick = (event, id, groupMethod, tool, layerCountInGroup) => {
+const onToolClick = (event, tool, group) => {
+    const id = group.getId();
+    const parentId = group.getParentId();
+    const groupMethod = group.getGroupMethod();
+    const layerCountInGroup = group.getLayers().length;
     const cb = tool.getCallback();
     if (cb) {
-        cb(event, id, groupMethod, layerCountInGroup);
+        cb(event, id, groupMethod, layerCountInGroup, parentId);
     }
     // Prevent collapse open on tool icon click
     event.stopPropagation();
 };
 
+const selectGroup = (event, checked, group, controller) => {
+    // if switch is checked, we add the groups layers to selected layers, if not, we remove all the layers from checked layers
+    !checked ? controller.addGroup(group) : controller.removeGroup(group);
+    event.stopPropagation();
+}
+
+// Confirm dialog hit cancel
+const deactivateGroup = (event, group, controller) => {
+    controller.deactivateGroup(group);
+    event.stopPropagation();
+}
+
+const onGroupSelect = (event, checked, group, controller) => { 
+    // check if we need to show warning (over 10 layers inside the group)
+    if(checked) {
+        if (!controller.showWarn(group)) {
+           event.stopPropagation();
+           return;
+        } else {
+            selectGroup(event, !checked, group, controller);
+        }
+    } else {
+        selectGroup(event, !checked, group, controller);
+    }
+};
+
+const SubCollapsePanel = ({ active, group, selectedLayerIds, selectedGroupIds, controller, showWarn, warnActive, propsNeededForPanel }) => {
+    const layerRows = group.getLayers().map((layer, index) => {
+        const layerProps = {
+            id: layer._id,
+            model: layer,
+            even: index % 2 === 0,
+            selected: Array.isArray(selectedLayerIds) && selectedLayerIds.includes(layer.getId()),
+            controller
+        };
+        return layerProps;
+    });
+
+    const badgeText = group.unfilteredLayerCount
+        ? layerRows.length + ' / ' + group.unfilteredLayerCount
+        : layerRows.length;
+
+    return (
+        <StyledSubCollapse>
+            <StyledSubCollapsePanel {...propsNeededForPanel}
+                header={group.getTitle()}
+                showArrow
+                extra={
+                    <React.Fragment>
+                            <Confirm
+                                title={<Message messageKey='grouping.manyLayersWarn'/>}
+                                visible={warnActive}
```

Comment:
The API in https://ant.design/components/popconfirm/ doesn't mention visible. What does it do? It has a disabled prop that you could use to toggle showing/not showing the confirmation. Also instead of passing warnActive through the handler could you just use `disabled={layers.length < 10}` or something like that?

Reply:
https://ant.design/components/popconfirm/ "Conditional trigger" example. I tried to do this in several other ways but it never worked properly with the switch.